storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -4214,6 +4203,22 @@ private void maybeAddPlaceholderBoltAggStats(TopologyPageInfo topoPageInfo, Stor
         }
     }
 
+    private CommonAggregateStats getPlaceholderCommonAggregateStats(Object component) {
+        // common aggregate
+        CommonAggregateStats commonStats = new CommonAggregateStats();
+
+        // set num_tasks and expected num_executors
+        Map<String, Object> jsonMap = StormCommon.componentConf(component);
+        int numTasks = (int) (long) jsonMap.getOrDefault(Config.TOPOLOGY_TASKS, 0);
+        commonStats.set_num_executors(numTasks);
```

Comment:
Just to double check, is it safe to assume number of executors is same as number of tasks? If there's safe way to get the value it would be better to apply that, otherwise 0 (even -1) might make less confusion.

Reply:
Num_executors does not have to be same as num_tasks. Changing num_executors to 0

Reply:
You can use `StormCommon.numStartExecutors` to get num_executors.

https://github.com/apache/storm/blob/master/storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java#L2392-L2405

Reply:
Thanks @Ethanlm. I'll pull in numExecutors from your suggestion.